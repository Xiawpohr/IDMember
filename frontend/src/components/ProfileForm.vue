<template>
  <p>profile form</p>
</template>

<script>
export default {
  props: {
    user: {
      types: Object,
      required: true,
      default: () => ({})
    }
  }
}
</script>
