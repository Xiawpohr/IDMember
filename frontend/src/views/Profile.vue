<template>
  <div>
    <h1>Your Profile</h1>
    <ProfileForm :user="currentUser"/>
  </div>
</template>

<script>
import ProfileForm from '@/components/ProfileForm.vue'

export default {
  components: {
    ProfileForm
  },
  data() {
    return {
      currentUser: {}
    }
  }
}
</script>
