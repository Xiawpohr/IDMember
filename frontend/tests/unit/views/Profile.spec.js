import { shallowMount, createLocalVue } from '@vue/test-utils'
import Profile from '@/views/Profile.vue'
import ProfileForm from '@/components/ProfileForm.vue'

const localVue = createLocalVue()

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Profile, {
    localVue
  })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should render ProfileForm component', () => {
  const profileForm = wrapper.find(ProfileForm)
  expect(profileForm.exists()).toBe(true)
})

it('should pass user prop to the ProfileForm component', () => {
  const user = { name: 'Arthur', gender: 'male' }
  const profileForm = wrapper.find(ProfileForm)
  wrapper.setData({ currentUser: user })
  expect(profileForm.vm.user).toEqual(user)
})
