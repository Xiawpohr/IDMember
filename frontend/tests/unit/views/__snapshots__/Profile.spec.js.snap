// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<div>
  <h1>Your Profile</h1>
  <profileform-stub user="[object Object]"></profileform-stub>
</div>
`;
